### ISSUE TITLE
[Bug] Inconsistent behavior in Site List entries

### ISSUE THREAD

**Issue by AliRezayiNejad** [darkreader/darkreader#12699]
### Prerequisites

- [X] I [searched for any existing report](https://github.com/darkreader/darkreader/issues?q=is%3Aissue) about this bug to avoid opening a duplicate.
- [X] I can reproduce this bug in a new, unmodified web browser profile with Dark Reader installed as the only extension.
- [X] I understand I need to use the [Broken Website Report template](https://github.com/darkreader/darkreader/issues/new?assignees=&labels=Broken+Website&template=broken-website-report.yml&title=%5BBroken+Website%5D+Replace+with+title) if this bug I am reporting occurs on a single website.

### Bug Description

(At least in "Invert listed only" mode,) Using *.example.com will affect www.example.com and any subdomains like sub1.example.com.
However, If a website does omit the www. part from it's URL (like archlinux.org), then the *.example.com format does not affect the site, a separate entry of example.com must be created for that.
This behavior is consistent on Firefox on multiple platform.

### Website Address

archlinux.org vs aur.archlinux.org or any other websites like it

### Steps To Reproduce

1. Switch to Invert listed only mode.
2. Add a URL like *.archlinux.org and another website that does not omit "www." like *.darebee.com to the site list.
3. Visit archlinux.org and darebee.com.

### Expected Behavior

Either the main domain of both websites has to be affected by Dark Reader, or neither. I personally prefer both.

### Actual Behavior

archlinux.org does is not affected by Dark Reader, and darebee.com is.
This behavior is consistent for all similar websites, based on how they handle the "www." in their URL.

### Operating System

Windows 10, Arch Linux, Android

### Web Browser name and version

Firefox 125.0.3

### Dark Reader version

4.9.84

**Comment by oberrich** [comment]
I have investigated this issue and unless I am misunderstanding, this is intended and semantically correct.
In order to match both `www.example.tld` as well as `example.tld` you can simply use `example.tld` and it will handle both domains correctly.

To match both `*.example.tld` as well as `example.tld` you have to explicitly add `example.tld`.
To match `www.example.tld` as well as `example.tld` you only need to specify `example.tld`.

**Comment by ahai64** [comment]
Yes, now my list is full of duplicate urls like `example.com` and `*.example.com`.
The way it used to work was more reasonable.

**Comment by realgooseman** [comment]
What would be a rule that would include the domain and subdomain ?

Like a single rule that would include `www.google.com` + `google.com` + `calendar.google.com`
For now I can make it work on Dark Reader but with two rules: `google.com` + `*.google.com`
